uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -0,0 +1,45 @@
+define([
+  'leave-absences/shared/modules/models-instances',
+  'leave-absences/shared/models/leave-status-id-model',
+  'common/models/instances/instance'
+], function (instances) {
+  'use strict';
+
+  instances.factory('LeaveRequestInstance', [
+    'ModelInstance',
+    'LeaveRequestAPI',
+    'LeaveStatusID',
+    function (ModelInstance, LeaveRequestAPI, LeaveStatusID) {
+      return ModelInstance.extend({
+
+        /**
+         * This method is used to cancel a leave request
+         *
+         * @return {Promise}
+         */
+        cancel: function () {
+          var leaveRequest = this;
+          return LeaveStatusID.getOptionIDByName("cancelled")
+            .then(function (cancelledStatusId) {
+              return leaveRequest.update({
+                'status_id': cancelledStatusId
+              })
+            })
+            .then(function (data) {
+              return data.values[0];
+            });
+        },
+
+        /**
+         * This method is used to cancel a leave request
+         *
+         * @param {object} attributes - Values which needs to be updated
+         * @return {Promise}
+         */
+        update: function (attributes) {
+          var leaveRequest = this;
```

Comment:
If i remember correctly, every `ModelInstance` has an `attributes()` method which returns a plain object with just the attributes, without the methods. Given that, I think you can do something like:
```js
update: function (newAttrs) {
  return LeaveRequestAPI.sendPOST('LeaveRequest', 'create', _.assign({}, this.attributes(), newAttrs));
}